pygmt/tests/test_grd2cpt.py

```diff
@@ -0,0 +1,342 @@
+"""
+Tests for grd2cpt.
+"""
+import os
+
+import numpy as np
+import pytest
+from pygmt import Figure, grd2cpt
+from pygmt.datasets import load_earth_relief
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import GMTTempFile
+from pygmt.helpers.testing import check_figures_equal
+
+TEST_DATA_DIR = os.path.join(os.path.dirname(__file__), "data")
+POINTS_DATA = os.path.join(TEST_DATA_DIR, "points.txt")
+
+
+@pytest.fixture(scope="module", name="points")
+def fixture_points():
+    """
+    Load the points data from the test file.
+    """
+    return np.loadtxt(POINTS_DATA)
+
+
+@pytest.fixture(scope="module", name="region")
+def fixture_region():
+    """
+    The data region.
+    """
+    return [10, 70, -5, 10]
+
+
+@pytest.fixture(scope="module", name="grid")
+def fixture_grid():
+    """
+    Load the grid data from the sample earth_relief file.
+    """
+    return load_earth_relief(registration="gridline")
+
+
+@check_figures_equal()
+def test_grd2cpt(grid):
+    """
+    Test creating a CPT with grd2cpt to create a CPT based off a grid input and
+    plot it with grdimage.
+    """
+    fig_ref, fig_test = Figure(), Figure()
+    # Use single-character arguments for the reference image
+    fig_ref.basemap(B="a", J="W0/15c", R="d")
+    grd2cpt(grid=grid)
+    fig_ref.grdimage(grid=grid)
+    fig_ref.colorbar(B="a2000")
+    fig_test.basemap(frame="a", projection="W0/15c", region="d")
+    grd2cpt(grid=grid)
+    fig_test.grdimage(grid=grid)
+    fig_test.colorbar(frame="a2000")
+    return fig_ref, fig_test
```

Comment:
Just one quick suggestion on unit testing. Best thing would be to **only** have `grd2cpt` in the test function and exclude other functions (e.g. `grdimage`, `basemap`, etc). That way our tests will run fast (`grdimage` can be slow) and changes other functions like `basemap` won't affect the `grd2cpt` test.

Personally would be ok to write the tests with just `colorbar` if possible (as with the newer `makecpt` tests written at https://github.com/GenericMappingTools/pygmt/blame/1303188ab7d777085a01b468606244e2f4f39d05/pygmt/tests/test_makecpt.py#L187-L222). But if there's a way to write a 'pure' `grd2cpt` only test using something like `output=temp.cpt`, then go for it!

Reply:
I removed `grdimage` from the tests, and it appears to be about 1 second faster to run the file (6 seconds vs. 7 seconds).